internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless.go

```diff
@@ -146,6 +148,51 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	return resourceRead(ctx, d, meta)
 }
 
+func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
+	projectID := d.Get("project_id").(string)
+	instanceName := d.Get("instance_name").(string)
+	endpointID := d.Get("endpoint_id").(string)
+
+	_, _, err := connV2.ServerlessPrivateEndpointsApi.GetServerlessPrivateEndpoint(ctx, projectID, instanceName, endpointID).Execute()
```

Comment:
what is this call for? we don't use any data from the response

Reply:
yes, it is needed because this resource is supposed to be used in conjunction with another resource `mongodbatlas_privatelink_endpoint_serverless` which creates the resource that is further updated by this service resource.
This actually should be more like a module but we have two resources instead.

cc @Zuhairahmed this is a good candidate for a module

Reply:
Great explanation Aastha - do we have this documented for customers? or is it intuitive when they are using it?

Reply:
so I am guessing you are using this get just to see if it exists? as Leo said, if the response is not used it's weird we are calling the get.

At minimum we should explain it in the code

Reply:
> this is more for error handling I'd say.

mmm what type of error handling? the only error you can get here is either the call fails for networking/atlas reasons (which means we are just making it worse) or because it doesn't exist..